[BUG] DP_DB_HOST env is missing in restore workloads

```
 ~  k describe pod restore-post-ready-2d130122-bp0-0-0-qt6m7                                                                                                     multi(k3d-multi/default)
Name:             restore-post-ready-2d130122-bp0-0-0-qt6m7
Namespace:        default
Priority:         0
Service Account:  kubeblocks-dataprotection-worker
Node:             k3d-multi-agent-1/172.21.0.6
Start Time:       Fri, 06 Dec 2024 15:39:22 +0800
Labels:           app.kubernetes.io/managed-by=kubeblocks-dataprotection
                  batch.kubernetes.io/controller-uid=1ebd9459-617e-4320-9958-52dae77db000
                  batch.kubernetes.io/job-name=restore-post-ready-2d130122-bp0-0-0
                  controller-uid=1ebd9459-617e-4320-9958-52dae77db000
                  dataprotection.kubeblocks.io/restore=banana98-postgresql-34a86de7-postready
                  job-name=restore-post-ready-2d130122-bp0-0-0
Annotations:      <none>
Status:           Failed
IP:               10.42.1.149
IPs:
  IP:           10.42.1.149
Controlled By:  Job/restore-post-ready-2d130122-bp0-0-0
Init Containers:
  dp-copy-datasafed:
    Container ID:  containerd://2cd53bad0b01f9297d01b0218e593b4a3e970dd6fc9eb4c1aa3ecef7f27c1816
    Image:         apecloud-registry.cn-zhangjiakou.cr.aliyuncs.com/apecloud/datasafed:0.2.0
    Image ID:      apecloud-registry.cn-zhangjiakou.cr.aliyuncs.com/apecloud/datasafed@sha256:ee48f0674aa4001ac0ce959c31f74dbe0159f5ddb317ef6a9f09454bdcfc418b
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/sh
      -c
      /scripts/install-datasafed.sh /bin/datasafed
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Fri, 06 Dec 2024 15:39:22 +0800
      Finished:     Fri, 06 Dec 2024 15:39:23 +0800
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     0
      memory:  0
    Requests:
      cpu:        0
      memory:     0
    Environment:  <none>
    Mounts:
      /bin/datasafed from dp-datasafed-bin (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-w57st (ro)
Containers:
  restore:
    Container ID:  containerd://8a1392f2b6f228e72361ad371e4ae878cad3a9c31a4c90b99cb055938402466f
    Image:         docker.io/apecloud/spilo:16.4.0
    Image ID:      docker.io/apecloud/spilo@sha256:ab7d6944172acd2257a94e2088dc42fb5e1965e5ce2c8aeb823cbf3ef9c1d707
    Port:          <none>
    Host Port:     <none>
    Command:
      bash
      -c
      # shellcheck disable=SC2148
      set -e
      set -o pipefail
      export PATH="$PATH:$DP_DATASAFED_BIN_PATH"
      export DATASAFED_BACKEND_BASE_PATH="$DP_BACKUP_BASE_PATH"
      export PGPASSWORD=${DP_DB_PASSWORD}
      function remote_file_exists() {
          local out=$(datasafed list $1)
          if [ "${out}" == "$1" ]; then
              echo "true"
              return
          fi
          echo "false"
      }

      if [ $(remote_file_exists "${DP_BACKUP_NAME}.zst") == "true" ]; then
        datasafed pull -d zstd-fastest "${DP_BACKUP_NAME}.zst" - | psql -U ${DP_DB_USER} -h ${DP_DB_HOST} -p ${DP_DB_PORT} -f ${DP_BACKUP_NAME}
        echo "restore complete!";
        exit 0
      fi

    State:          Terminated
      Reason:       Error
      Exit Code:    2
      Started:      Fri, 06 Dec 2024 15:39:24 +0800
      Finished:     Fri, 06 Dec 2024 15:39:24 +0800
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     0
      memory:  0
    Requests:
      cpu:     0
      memory:  0
    Environment Variables from:
      banana98-postgresql-env  ConfigMap  Optional: false
    Environment:
      DP_BACKUP_NAME:                bp0
      DP_BACKUP_BASE_PATH:           /default/pg1-3d4fd774-2201-4f98-b21d-4f1c993d6ede/postgresql/bp0
      DP_BACKUP_STOP_TIME:           2024-12-06T07:26:51Z
      IMAGE_TAG:                     16.4.0
      POSTGRES_USER:                 <set to the key 'username' in secret 'banana98-postgresql-account-postgres'>  Optional: false
      POSTGRES_PASSWORD:             <set to the key 'password' in secret 'banana98-postgresql-account-postgres'>  Optional: false
      DCS_ENABLE_KUBERNETES_API:     true
      KUBERNETES_USE_CONFIGMAPS:     true
      SCOPE:                         $(POSTGRES_COMPONENT_NAME)
      KUBERNETES_SCOPE_LABEL:        apps.kubeblocks.postgres.patroni/scope
      KUBERNETES_ROLE_LABEL:         apps.kubeblocks.postgres.patroni/role
      KUBERNETES_LABELS:             {"app.kubernetes.io/managed-by":"kubeblocks","app.kubernetes.io/instance":"$(CLUSTER_NAME)","apps.kubeblocks.io/component-name":"$(POSTGRES_COMPONENT_SHORT_NAME)","apps.kubeblocks.postgres.patroni/scope":"$(POSTGRES_COMPONENT_NAME)"}
      RESTORE_DATA_DIR:              /home/postgres/pgdata/kb_restore
      KB_PG_CONFIG_PATH:             /home/postgres/conf/postgresql.conf
      SPILO_CONFIGURATION:           bootstrap:
                                       initdb:
                                         - auth-host: md5
                                         - auth-local: trust

      ALLOW_NOSSL:                   true
      PGROOT:                        /home/postgres/pgdata/pgroot
      PGUSER_SUPERUSER:              $(POSTGRES_USER)
      PGPASSWORD_SUPERUSER:          $(POSTGRES_PASSWORD)
      PGUSER_ADMIN:                  superadmin
      PGPASSWORD_ADMIN:              $(POSTGRES_PASSWORD)
      PGUSER_STANDBY:                standby
      PGPASSWORD_STANDBY:            $(POSTGRES_PASSWORD)
      PGUSER:                        $(POSTGRES_USER)
      PGPASSWORD:                    $(POSTGRES_PASSWORD)
      CURRENT_POD_NAME:              restore-post-ready-2d130122-bp0-0-0-qt6m7 (v1:metadata.name)
      CURRENT_POD_IP:                 (v1:status.podIP)
      CURRENT_POD_HOST_IP:            (v1:status.hostIP)
      POD_IP:                        $(CURRENT_POD_IP)
      POD_NAMESPACE:                 $(CLUSTER_NAMESPACE)
      DP_DB_USER:                    <set to the key 'username' in secret 'pg1-postgresql-account-postgres'>  Optional: false
      DP_DB_PASSWORD:                <set to the key 'password' in secret 'pg1-postgresql-account-postgres'>  Optional: false
      DP_DB_PORT:                    5432
      DATASAFED_LOCAL_BACKEND_PATH:  /backupdata
      DP_DATASAFED_BIN_PATH:         /bin/datasafed
    Mounts:
      /backupdata from dp-backup-data (rw)
      /bin/datasafed from dp-datasafed-bin (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-w57st (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  dp-backup-data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  pvc-my-repo-b22jjm
    ReadOnly:   false
  dp-datasafed-bin:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  kube-api-access-w57st:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:                      <none>
➜  ~

--- a/pkg/dataprotection/restore/builder.go
+++ b/pkg/dataprotection/restore/builder.go
@@ -253,11 +253,22 @@ func (r *restoreJobBuilder) addTargetPodAndCredentialEnv(pod *corev1.Pod,
 		env = pod.Spec.Containers[0].Env
 		r.envFrom = pod.Spec.Containers[0].EnvFrom
 	}
-	env = append(env, corev1.EnvVar{Name: dptypes.DPDBHost, Value: intctrlutil.BuildPodHostDNS(pod)})
-	if connectionCredential == nil {
-		if portEnv, err := utils.GetDPDBPortEnv(pod, target.ContainerPort); err == nil {
+	addDBHostEnv := func() {
+		env = append(env, corev1.EnvVar{Name: dptypes.DPDBHost, Value: intctrlutil.BuildPodHostDNS(pod)})
+	}
+	addDBPortEnv := func() {
+		portEnv, err := utils.GetDPDBPortEnv(pod, target.ContainerPort)
+		if err != nil {
+			// fallback to use the first port of the pod
+			portEnv, _ = utils.GetDPDBPortEnv(pod, nil)
+		}
+		if portEnv != nil {
 			env = append(env, *portEnv)
 		}
+	}
+	if connectionCredential == nil {
+		addDBHostEnv()
+		addDBPortEnv()
 	} else {
 		appendEnvFromSecret := func(envName, keyName string) {
 			if keyName == "" {
@@ -277,11 +288,12 @@ func (r *restoreJobBuilder) addTargetPodAndCredentialEnv(pod *corev1.Pod,
 		if connectionCredential.PortKey != "" {
 			appendEnvFromSecret(dptypes.DPDBPort, connectionCredential.PortKey)
 		} else {
-			portEnv, _ := utils.GetDPDBPortEnv(pod, nil)
-			env = append(env, *portEnv)
+			addDBPortEnv()
 		}
 		if connectionCredential.HostKey != "" {
 			appendEnvFromSecret(dptypes.DPDBHost, connectionCredential.HostKey)
+		} else {
+			addDBHostEnv()
 		}
 	}
 	r.env = utils.MergeEnv(r.env, env)

--- a/pkg/dataprotection/restore/builder.go
+++ b/pkg/dataprotection/restore/builder.go
@@ -253,8 +253,8 @@ func (r *restoreJobBuilder) addTargetPodAndCredentialEnv(pod *corev1.Pod,
 		env = pod.Spec.Containers[0].Env
 		r.envFrom = pod.Spec.Containers[0].EnvFrom
 	}
+	env = append(env, corev1.EnvVar{Name: dptypes.DPDBHost, Value: intctrlutil.BuildPodHostDNS(pod)})
 	if connectionCredential == nil {
-		env = append(env, corev1.EnvVar{Name: dptypes.DPDBHost, Value: intctrlutil.BuildPodHostDNS(pod)})
 		if portEnv, err := utils.GetDPDBPortEnv(pod, target.ContainerPort); err == nil {
 			env = append(env, *portEnv)
 		}

